epilogue-processor/src/main/java/edu/wpi/first/epilogue/processor/EpiloguerGenerator.java

```diff
@@ -0,0 +1,172 @@
+        out.println();
+
+        loggerClassNames.forEach(
+            name -> {
+              String simple = StringUtils.simpleName(name);
+
+              // public static final FooLogger fooLogger = new FooLogger();
+              out.print("  public static final ");
+              out.print(simple);
+              out.print(" ");
+              out.print(StringUtils.lowerCamelCase(simple));
+              out.print(" = new ");
+              out.print(simple);
+              out.println("();");
+            });
+        m_customLoggers.values().stream()
+            .distinct()
+            .forEach(
+                loggerType -> {
+                  var loggerTypeName = loggerType.asElement().getSimpleName();
+                  out.println(
+                      "  public static final "
+                          + loggerTypeName
+                          + " "
+                          + StringUtils.lowerCamelCase(loggerTypeName)
+                          + " = new "
+                          + loggerTypeName
+                          + "();");
+                });
+        out.println();
+
+        out.println(
+            """
+              public static void configure(java.util.function.Consumer<EpilogueConfiguration> configurator) {
+                configurator.accept(config);
+              }
+
+              public static EpilogueConfiguration getConfig() {
+                return config;
+              }
+            """);
+
+        out.println(
+            """
+              /**
+               * Checks if data associated with a given importance level should be logged.
+               */
+              public static boolean shouldLog(Epilogue.Importance importance) {
+                return importance.compareTo(config.minimumImportance) >= 0;
+              }
+            """
+                .stripTrailing());
+
+        // Only generate a binding if the robot class is a TimedRobot
+        if (!mainRobotClasses.isEmpty()) {
+          for (TypeElement mainRobotClass : mainRobotClasses) {
+            String robotClassName = mainRobotClass.getQualifiedName().toString();
+
+            out.println();
+            out.print(
+                """
+                  /**
+                   * Binds Epilogue updates to a timed robot's update period. Log calls will be made at the
+                   * same update rate as the robot's loop function, but will be offset by a full phase
+                   * (for example, a 20ms update rate but 10ms offset from the main loop invocation) to
+                   * help avoid high CPU loads. However, this does mean that any logged data that reads
+                   * directly from sensors will be slightly different from data used in the main robot
+                   * loop.
+                   */
+                """);
+            out.println("  public static void bind(" + robotClassName + " robot) {");
+            out.println("    robot.addPeriodic(() -> {");
```

Comment:
This is why the robot class should be checked for TimedRobot. `addPeriodic` is only defined on that subclass, not on RobotBase